fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBModificationQueryTest.java

```diff
@@ -207,31 +214,33 @@ public void testPlanInsertExpression() throws Exception {
                     EvaluationContext.empty()).getPlan();
 
             assertMatchesExactly(plan, insertPlan(explodePlan()).where(target(equalsObject("RestaurantRecord"))));
-
-            var resultValues = fetchResultValues(context, plan, record -> {
-                final var recordDescriptor = record.getDescriptorForType();
-                final var rest_no = recordDescriptor.findFieldByName("rest_no");
-                final var name = recordDescriptor.findFieldByName("name");
-                switch ((int)(long)record.getField(rest_no)) {
-                    case 100:
-                        Assertions.assertEquals("Burger King", record.getField(name));
-                        break;
-                    case 200:
-                        Assertions.assertEquals("Heirloom Cafe", record.getField(name));
-                        break;
-                    default:
-                        Assertions.fail("unexpected record");
-                }
-                return record;
-            }, c -> { });
-            Assertions.assertEquals(2, resultValues.size());
+            for (ExecuteProperties p : List.of(ExecuteProperties.newBuilder().setDryRun(true).build(), ExecuteProperties.SERIAL_EXECUTE)) {
```

Comment:
If I'm reading this correctly, this doesn't actually assert that the update doesn't happen if `dryRun` mode is true. It would be nice if, for example, the `selectPlan` were executed after each execution, and it validated that the select plan returned no results when `dryRun` was `true`.